### ISSUE TITLE
Experimental feature failure!

### ISSUE THREAD

**Issue by awuetz** [hashicorp/terraform#10024]
Hi,

i just got this failure after changing something at the output variables of my module "vpc".

### Terraform Version
 terraform -v
Terraform v0.7.9

### Terraform output
```
Experimental feature failure! Please report a bug.

This is not an error. Your Terraform operation completed successfully.
Your real infrastructure is unaffected by this message.

While running, Terraform sometimes tests experimental features in the
background. These features cannot affect real state and never touch
real infrastructure. If the features work properly, you see nothing.
If the features fail, this message appears.

The following failures happened while running experimental features.
Please report a Terraform bug so that future Terraform versions that
enable these features can be improved!

You can report an issue at: https://github.com/hashicorp/terraform/issues

1 error(s) occurred:

* apply operation: Real and shadow states do not match! Real state:

<no state>
module.vpc:
  <no state>
  Outputs:

  01 Environment = POC
  01_Environment = POC
  02 VPC Netrange = 10.1.0.0/16
  02_VPC_Netrange = 10.1.0.0/16
  03 Webserver public instance_ips = [35.156.68.x]
  03_Webserver_public_instance_ips = [35.156.0.x]
  04 Appserver public instance_ips = []
  04_Appserver_public_instance_ips = []
  Appserver public instance_ips = []
  Environment = POC
  VPC Netrange = 10.1.0.0/16
  Webserver public instance_ips = [35.156.0.116]
  instance_ips = [35.156.29.49]
  vpc cidr_block = 10.1.0.0/16

Shadow state:

<no state>
module.vpc:
  <no state>

This is not an error. Your terraform operation completed successfully
and your real infrastructure is unaffected by this message.

```

### Affected Resource(s) and steps to reproduce the issue:
I changed the outputs.tf file (Commented out all the output variables) in my module "vpc". I could additionally observe, that i got the same error as i changed the outputs NAME from e.g. 01 Environment to 01_Environment

../modules/
└── vpc
    ├── outputs.tf

1. `terraform apply`

Greetings Alex

**Comment by mitchellh** [comment]
This is fixed in 0.7.10 and was caused by outputs not being pruned in the old state. At least, based on your description I'm confident this is the case. If this isn't please reopen the issue!

**Comment by awuetz** [comment]
Thanks i will try it. Is the 0.7.10 already released? Because on the website 0.7.9 is the newest zip file i can find..

**Comment by mitchellh** [comment]
@awuetz You might be seeing a cache, it was released a couple days ago, shows 0.7.10 for me
